src/geometry/util.rs

```diff
@@ -28,7 +28,7 @@ impl Transformation {
 
 // Distributor ///////////////////////////////////////////////////////////////
 //
-// A helper object to distribute bodies in space with velcoity. The
+// A helper object to distribute bodies in space with velocity. The
 // distribution is uses parameterized randomization.
 
 pub struct Distributor {
```

Comment:
if u have time, could you maybe explain for me the code of Distributor? 
for ex. `angle_rand = thread_rng()`what is it for ? 
`let position = &trans * Vector` and so on ....

Reply:
Sure, the idea of the `Distributor` is to generate a certain number of bodies in space. At the moment I've only defined one type of distribution, which is a _radial (circular)_ distribution. The way it works is to imagine a coordinate system where the origin (0, 0) is at the center. Then we create a vector extending from the origin and along the x axis in the positive direction. The length of this vector is `dist`, and it determines how far away the body will be from the origin. We also generate a random angle, between 0 and 360 degrees (0 and 2PI radians). This is used to rotate the vector (which is pointing due east) around the origin. Then, we can use this vector do define the position of the body.

You may notice that the body has a velocity too. Before the vector is rotated, the velocity of the vector is pointing due north. We also rotate the velocity by the same amount that we rotate the initial vector, so the end result is that the body is positioned randomly around the center, and it has a velocity that is pointing along the tangent of the circle centered on the origin.

Reply:
`angle_rand` is just a random number generator to produce random angles. `&trans * Vector ...` means we _transform_ the vector (in this case `trans` is a rotation transformation/matrix).